internal/injector/aspect/context/golang.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package context
+
+import (
+	"fmt"
+	"go/version"
+
+	"gopkg.in/yaml.v3"
+)
+
+// GoLang represents a go language level. It's a string of the form "go1.18".
+type GoLang struct {
```

Comment:
Would you mind adding a "version" in there? Like `GoVersion` or `GoLanguageVersion` ? The first read of this PR was verry confusing

Reply:
I agree; but since it's internal anyway, it's okay for it to be there for now... Would move it later if needed.